packages/lsd-react/src/components/PortalProvider/usePortal.ts

```diff
@@ -13,7 +13,14 @@ export const usePortal = ({ parentId }: Props) => {
 
   useEffect(() => {
     if (typeof window === 'undefined' || !elementRef.current) return
-    document.getElementById(parentId)?.appendChild(elementRef.current)
+
+    const parentElements = document.querySelectorAll(`#${parentId}`)
+
+    // In some places (e.g. storybook), there may be multiple portal containers when a component
+    // is rendered multiple times. Here, we append only to the last found parent element.
+    // This is because usually the last parent is the most recently rendered one.
+    // Without this, we may append to a parent that is about to be removed from the DOM.
+    parentElements[parentElements.length - 1]?.appendChild(elementRef.current)
```

Comment:
Main issue here were 2 lsd-presentation containers were present, and we were calling appendChild on the container that was about to be removed from the DOM. After it was removed from the DOM, the dropdown / calendar would not open. This happens on storybook - not sure if it happens in other places. Here's a screenshot with 2 lsd-presentation elements:

![Screenshot from 2023-10-19 13-19-23](https://github.com/acid-info/lsd/assets/9993816/ba46e99c-a63f-46fc-8db9-cdffc0975056)